UnityProject/Assets/Scripts/Tilemaps/Tiles/BasicTile.cs

```diff
@@ -59,4 +85,139 @@ public bool IsSpace()
+
+		}
+		else if (DeconstructionType == DeconstructionType.Wirecutters)
+		{
+			tileChangeManager.RemoveTile(Vector3Int.RoundToInt(cellPos), tile.LayerType);
+			SoundManager.PlayNetworkedAtPos("WireCutter", interaction.WorldPositionTarget, Random.Range(0.8f, 1.2f));
+			tileChangeManager.SubsystemManager.UpdateAt(cellPos);
+			DoSpawnOnDeconstruct(interaction);
+		}
+		else if (DeconstructionType == DeconstructionType.Wrench)
+		{
+			tileChangeManager.RemoveTile(Vector3Int.RoundToInt(cellPos), tile.LayerType);
+			SoundManager.PlayNetworkedAtPos("Wrench", interaction.WorldPositionTarget, Random.Range(0.8f, 1.2f));
+			tileChangeManager.SubsystemManager.UpdateAt(cellPos);
+			DoSpawnOnDeconstruct(interaction);
+		}
+		else if (DeconstructionType == DeconstructionType.NormalWall)
+		{
+			//unweld to a girder
+			var progressFinishAction = new ProgressCompleteAction(
+				() =>
+				{
+					SoundManager.PlayNetworkedAtPos("Weld", interaction.WorldPositionTarget, 0.8f);
+					tileChangeManager.RemoveTile(cellPos, LayerType.Walls);
+					SoundManager.PlayNetworkedAtPos("Deconstruct", interaction.WorldPositionTarget, 1f);
+
+					//girder / metal always appears in place of deconstructed wall
+					Spawn.ServerPrefab(CommonPrefabs.Instance.Girder, interaction.WorldPositionTarget);
+					Spawn.ServerPrefab(CommonPrefabs.Instance.Metal, interaction.WorldPositionTarget);
+					tileChangeManager.SubsystemManager.UpdateAt(cellPos);
+				}
+			);
+
+			//Start the progress bar:
+			var bar = UIManager.ServerStartProgress(ProgressAction.Construction, interaction.WorldPositionTarget,
+				10f, progressFinishAction, interaction.Performer);
+			if (bar != null)
+			{
+				SoundManager.PlayNetworkedAtPos("Weld", interaction.WorldPositionTarget, Random.Range(0.9f, 1.1f));
+			}
+		}
+		//TODO: Deconstruct reinforced wall, will spawn a prefab to handle deconstruction, not yet implemented
+	}
+
+	private void DoSpawnOnDeconstruct(PositionalHandApply interaction)
+	{
+		if (SpawnOnDeconstruct != null)
+		{
+			//always spawn at least one floor tile.
+			var amount = TileType == TileType.Floor ? Math.Max(1, spawnAmountOnDeconstruct) : spawnAmountOnDeconstruct;
+        	Spawn.ServerPrefab(SpawnOnDeconstruct, interaction.WorldPositionTarget, count: amount);
+        }
+	}
+
+	private GameObject DefaultSpawnOnDeconstruct()
+	{
+		if (LayerType == LayerType.Floors)
+		{
+			return CommonPrefabs.Instance.FloorTile;
+		}
+
+		return null;
+	}
+
+
+	private DeconstructionType DefaultDeconstructionType()
+	{
+		if (LayerType == LayerType.Floors)
+		{
+			return DeconstructionType.Crowbar;
+		}
+
+		return DeconstructionType.None;
+	}
+}
+
+/// <summary>
+/// Identifies the deconstruction logic that should be used for this tile
+/// </summary>
+public enum DeconstructionType
+{
+	//can't deconstruct
+	None = 0,
+	//pry off with crowbar
+	Crowbar = 1,
+	//snip with wirecutters
+	Wirecutters = 2,
+	//dismantle with wrench
+	Wrench = 3,
+	//cut with to turn into girder
```

Comment:
```suggestion
	//weld to turn into girder
```

Reply:
is `DeconstructionType` just for one step? if so, `NormalWall` can be renamed to `Weld`

Reply:
It's not just for one step or I would've called it weld.